scipy/linalg/_basic.py

```diff
@@ -1049,9 +1087,9 @@ def det(a, overwrite_a=False, check_finite=True):
     # Empty array has determinant 1 because math.
     if min(*a1.shape) == 0:
         if a1.ndim == 2:
-            return np.float64(1.)
+            return a1.dtype.type(1.)
```

Comment:
Related to the `xfail` comment: this change and the one two lines below should probably be reverted, since `float64` was intended.